## Teste de Duas Variâncias

8 amostras foram tomadas de uma máquina A e o desvio padrão observado foi 1,1. De outra máquina B tomou-se 5 amostras e a variância observada foi 11. Há diferença entre as variâncias das máquinas A e B? Nível de confiança: 90%.

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import scipy.stats as stats
from scipy.stats import f

In [ ]:

F_cal = 11/(1.1**2)
F_cal

9.09090909090909

Agora temos que encontrar o valor crítico à direita e à esquerda. Para tanto, usaremos a função isf da distribuição f, com os parâmetros 0.05 (alpha/2) e os dfs 4 e 7.

In [4]:
f.isf(0.05, 4, 7)

4.120311726897633

In [5]:
f.isf(0.95, 4, 7)

0.1640901524729093

Como 9.09 é maior que 4.12 (cauda direita), rejeitamos a hipótese nula. Portanto, a variância das duas máquinas é diferente.

Há uma forma de realizarmos o teste de hipótese diretamente?

In [7]:
m1 = [150, 150, 151, 148, 151, 152, 149, 150]
m2 = [135, 152, 154, 169, 157]

In [8]:
stats.levene(m1, m2)

LeveneResult(statistic=5.416245804750017, pvalue=0.040060452607943706)

Temos que alpha = 0.1 \
Como P Value < alpha, rejeitamos a hipótese nula. Portanto, a variância das duas máquinas é diferente.